zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/VersionManager.java

```diff
@@ -155,5 +155,16 @@ public Optional<Integer> findResourceVersionBefore(
     return getVersionInfo().findVersionBefore(version);
   }
 
+  public void forEachResource(final ResourceVisitor resourceVisitor) {
+    versionInfoColumnFamily.forEach(
+        versionInfo ->
+            resourceVisitor.visit(
+                idKey.getBuffer(), tenantAwareIdKey.tenantKey().toString(), versionInfo));
+  }
+
   private record TenantIdAndResourceId(String tenantId, String resourceId) {}
+
+  public interface ResourceVisitor {
+    boolean visit(DirectBuffer resourceId, String tenantId, VersionInfo versionInfo);
```

Comment:
:upside_down_face: Let's convert the resource id to string too, we already do it for the tenant id

Reply:
I assume `tenantId` was added later so it was preferred to use `String` everywhere :shrug:? 
I used the same type so there is no need to copy/allocate/convert :+1: